<a href="https://colab.research.google.com/github/Jkhou1000/Project-2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
#Basic 4
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.pipeline import make_pipeline
from sklearn.compose import ColumnTransformer

#import accuracy, precision, recall, classification report, and confusion matrix scoring functions
from sklearn.metrics import accuracy_score, precision_score, recall_score, classification_report, ConfusionMatrixDisplay, confusion_matrix


In [ ]:
df1 = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Stack 2/Week 7/Project 2/CSVS/mushrooms.csv')

df1.head()
df2 = pd.read_csv('/content/drive/MyDrive/Coding Dojo/Stack 2/Week 7/Project 2/CSVS/hippoCorpusV2.csv')


In [ ]:
df1.shape

(8124, 23)

In [ ]:
df1.head()

class cap-shape cap-surface cap-color bruises odor gill-attachment  \
0     p         x           s         n       t    p               f   
1     e         x           s         y       t    a               f   
2     e         b           s         w       t    l               f   
3     p         x           y         w       t    p               f   
4     e         x           s         g       f    n               f   

  gill-spacing gill-size gill-color  ... stalk-surface-below-ring  \
0            c         n          k  ...                        s   
1            c         b          k  ...                        s   
2            c         b          n  ...                        s   
3            c         n          n  ...                        s   
4            w         b          k  ...                        s   

  stalk-color-above-ring stalk-color-below-ring veil-type veil-color  \
0                      w                      w         p          w   
1                      w                      w         p          w   
2                      w                      w         p          w   
3                      w                      w         p          w   
4                      w                      w         p          w   

  ring-number ring-type spore-print-color population habitat  
0           o         p                 k          s       u  
1           o         p                 n          n       g  
2           o         p                 n          n       m  
3           o         p                 k          s       u  
4           o         e                 n          a       g  

[5 rows x 23 columns]

Source: UCI Machine LEarning respository Mushrooms

Description: a dataset involving the features of mushrooms which will determines its features and characteristics. 

Target: e or p, edible or poisonous.

this is a classification problem. the goal and target is to make a model that accurately and precisely predicts whether or not a mushroom is edible or poisonous for people who gather mushrooms. 

there are 8124 rows and 23 columns. The amount of rows is managable, but the major issue I foresee is working with the values. They're all single letters which are simple and clean but they're hard to figure out without referring to the data dictionary, and replacing all of them manually would be time consuming

Another issue its decinding which data matters for the prediction model. a lot of the features seem less important but a small difference can be important in identifying different species.



In [ ]:
df2.shape

(6854, 23)

In [ ]:
df2.head()

AssignmentId  WorkTimeInSeconds  WorkerId  annotatorAge  \
0  32RIADZISTQWI5XIVG5BN0VMYFRS4U               1641  XI8VK89S          25.0   
1  3018Q3ZVOJCZJFDMPSFXATCQ4DARA2               1245  1HN5ZZ1D          25.0   
2  3IRIK4HM3B6UQBC0HI8Q5TBJZLEC61               1159  8SBPL7EI          35.0   
3  3018Q3ZVOJCZJFDMPSFXATCQG04RAI                500  M1QQED2V          30.0   
4  3MTMREQS4W44RBU8OMP3XSK8NMJAWZ               1074  DU3RPZDB          25.0   

  annotatorGender annotatorRace  distracted  draining  frequency  importance  \
0             man         white         1.0       1.0        NaN         3.0   
1           woman         white         1.0       1.0        3.0         4.0   
2           woman         black         1.0       1.0        NaN         4.0   
3           woman         white         1.0       4.0        3.0         5.0   
4             man         white         2.0       2.0        NaN         3.0   

   ...                                     mostSurprising openness  \
0  ...                         when I got concert tickets     0.00   
1  ...                          we saw the beautiful sky.     1.00   
2  ...                          she went into labor early     0.50   
3  ...                     finding out they were healthy.     1.00   
4  ...  When I don't answer the phone in case I owe th...     0.25   

  recAgnPairId                    recImgPairId  similarity  \
0          NaN  3018Q3ZVOJCZJFDMPSFXATCQ4DARA2         3.0   
1          NaN  3018Q3ZVOJCZJFDMPSFXATCQ4DARA2         NaN   
2          NaN  3018Q3ZVOJCZJFDMPSFXATCQG04RAI         3.0   
3          NaN  3018Q3ZVOJCZJFDMPSFXATCQG04RAI         NaN   
4          NaN  3018Q3ZVOJCZJFDMPSFXATCQG06AR3         4.0   

                                similarityReason  \
0  I've been to a couple concerts, but not many.   
1                                            NaN   
2                           I am a mother myself   
3                                            NaN   
4             Because I also have money problems   

                                               story  stressful  \
0  Concerts are my most favorite thing, and my bo...        1.0   
1  The day started perfectly, with a great drive ...        1.0   
2  It seems just like yesterday but today makes f...        1.0   
3  Five months ago, my niece and nephew were born...        2.0   
4  About a month ago I went to burning man. I was...        4.0   

                                             summary timeSinceEvent  
0  My boyfriend and I went to a concert together ...           90.0  
1  My boyfriend and I went to a concert together ...           90.0  
2  My sister gave birth to my twin niece and neph...          150.0  
3  My sister gave birth to my twin niece and neph...          150.0  
4  It is always a journey for me to go to burning...           30.0  

[5 rows x 23 columns]

Source: https://www.kaggle.com/datasets/manishkc06/human-memory-and-cognition

Descrption: A dataset looking at the conginitive process of humans based on their ability to tell a story based on imagination and memory

Target: memtype, Type of story (imagined, recalled, or retold)

This is another classification problem. It attempts to figure out if a story is imagined, or if its directly recalled or retold differently. There are 6854 rows and 23 features.

I think the major issue I'd have is sorting out the missing data properly. there is a noticeable amount of NaN viewable from the head() alone. There's also the matter of which features to train the data on, there are a lot of unique values since they comprise of unique sentences and this study works off of peoples ability to tell a unique and imaginative story. 

Attribute Information: (classes: edible=e, poisonous=p)

* cap-shape: bell=b,conical=c,convex=x,flat=f, knobbed=k,sunken=s

* cap-surface: fibrous=f,grooves=g,scaly=y,smooth=s

* cap-color: brown=n,buff=b,cinnamon=c,gray=g,green=r,pink=p,purple=u,red=e,white=w,yellow=y

* bruises: bruises=t,no=f

* odor: almond=a,anise=l,creosote=c,fishy=y,foul=f,musty=m,none=n,pungent=p,spicy=s

* gill-attachment: attached=a,descending=d,free=f,notched=n

* gill-spacing: close=c,crowded=w,distant=d

* gill-size: broad=b,narrow=n

* gill-color: black=k,brown=n,buff=b,chocolate=h,gray=g, green=r,orange=o,* pink=p,purple=u,red=e,white=w,yellow=y

* stalk-shape: enlarging=e,tapering=t

* stalk-root: bulbous=b,club=c,cup=u,equal=e,rhizomorphs=z,rooted=r,missing=?

* stalk-surface-above-ring: fibrous=f,scaly=y,silky=k,smooth=s

* stalk-surface-below-ring: fibrous=f,scaly=y,silky=k,smooth=s

* stalk-color-above-ring: brown=n,buff=b,cinnamon=c,gray=g,orange=o,pink=p,red=e,white=w,yellow=y

* stalk-color-below-ring: brown=n,buff=b,cinnamon=c,gray=g,orange=o,pink=p,red=e,white=w,yellow=y

* veil-type: partial=p,universal=u

* veil-color: brown=n,orange=o,white=w,yellow=y

* ring-number: none=n,one=o,two=t

* ring-type: cobwebby=c,evanescent=e,flaring=f,large=l,none=n,pendant=p,sheathing=s,zone=z

* spore-print-color: black=k,brown=n,buff=b,chocolate=h,green=r,orange=o,purple=u,white=w,yellow=y

* population: abundant=a,clustered=c,numerous=n,scattered=s,several=v,solitary=y

* habitat: grasses=g,leaves=l,meadows=m,paths=p,urban=u,waste=w,woods=d

AssignmentId: Unique ID of this story

WorkTimeInSeconds: Time in seconds that it took the worker to do the entire HIT (reading instructions, story writing, questions)

WorkerId: Unique ID of the worker (random string, not MTurk worker ID)

annotatorAge: Lower limit of the age bucket of the worker. Buckets are: 18-24, 25-29, 30-34, 35-39, 40-44, 45-49, 50-54, 55+

annotatorGender: Gender of the worker

annotatorRace: Race/ethnicity of the worker

distracted: How distracted were you while writing your story? (5-point Likert)

draining: How taxing/draining was writing for you emotionally? (5-point Likert)

frequency: How often do you think about or talk about this event? (5-point Likert)

importance: How impactful, important, or personal is this story/event to you? (5-point Likert)

logTimeSinceEvent: Log of time (days) since the recalled event happened

mainEvent: Short phrase describing the main event described

memType: Type of story (recalled, imagined, retold) - The target variable

mostSurprising: Short phrase describing what the most surprising aspect of the story was

openness: Continuous variable representing the openness to experience of the worker

recAgnPairId: ID of the recalled story that corresponds to this retold story (null for imagined stories). Group on this variable to get the recalled-retold pairs.

recImgPairId: ID of the recalled story that corresponds to this imagined story (null for retold stories). Group on this variable to get the recalled-imagined pairs.

similarity: How similar to your life does this event/story feel to you? (5-point Likert)

similarityReason: Free text annotation of similarity

story: Story about the imagined or recalled event (15-25 sentences)

stressful: How stressful was this writing task? (5-point Likert)

summary: Summary of the events in the story (1-3 sentences)

timeSinceEvent: Time (number of days) since the recalled event happened